front/app/components/SignUpIn/SignUpInModal.tsx

```diff
@@ -85,15 +78,28 @@ const SignUpInModal = memo<Props>(
 
       return () =>
         subscriptions.forEach((subscription) => subscription.unsubscribe());
-      // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [authUser]);
 
-    const onClose = () => {
-      closeSignUpInModal();
+    const onClose = async () => {
+      const signedUpButNotCompleted =
+        !isNilOrError(authUser) &&
+        !authUser.attributes.registration_completed_at;
+
+      if (signedUpButNotCompleted) {
+        // We need to await signOut. If authUser would be there
+        // when we call closeSignUpModal,
+        // it would cause openSignUpInModalIfNecessary in App/index.tsx to open the modal again.
+        // This happens because the user is indeed not completely registered/verified
+        // (see openSignUpInModalIfNecessary).
+        await signOut();
+      }
 
       if (onDeclineInvitation && metaData?.isInvitation) {
         onDeclineInvitation();
       }
+
+      trackEventByName(tracks.signUpFlowExitedAtEmailVerificationStep);
```

Comment:
Do we want to move this up to the if block on line 88? Right now it will also fire if the invitation is declined, which I think is a different flow?